geoengine/permissions.py

```diff
@@ -0,0 +1,253 @@
+
+import requests as req
+from geoengine import api
+
+from geoengine.auth import get_session
+from geoengine.datasets import DatasetId
+from geoengine.error import GeoEngineException
+from geoengine.layers import LayerCollectionId, LayerId
+
+
+class RoleId:
+    '''A wrapper for a role id'''
+
+    def __init__(self, role_id: UUID) -> None:
+        self.__role_id = role_id
+
+    @classmethod
+    def from_response(cls, response: Dict[str, str]) -> RoleId:
+        '''Parse a http response to an `RoleId`'''
+
+        if 'id' not in response:
+            raise GeoEngineException(response)
+
+        role_id = response['id']
+
+        return RoleId(UUID(role_id))
+
+    def __eq__(self, other) -> bool:
+        '''Checks if two role ids are equal'''
+        if not isinstance(other, self.__class__):
+            return False
+
+        return self.__role_id == other.__role_id  # pylint: disable=protected-access
+
+    def __str__(self) -> str:
+        return str(self.__role_id)
+
+    def __repr__(self) -> str:
+        return repr(self.__role_id)
+
+
+class UserId:
+    '''A wrapper for a role id'''
+
+    def __init__(self, user_id: UUID) -> None:
+        self.__user_id = user_id
+
+    @classmethod
+    def from_response(cls, response: Dict[str, str]) -> UserId:
+        '''Parse a http response to an `UserId`'''
+        print(response)
+        if 'id' not in response:
+            raise GeoEngineException(response)
+
+        user_id = response['id']
+
+        return UserId(UUID(user_id))
+
+    def __eq__(self, other) -> bool:
+        '''Checks if two role ids are equal'''
+        if not isinstance(other, self.__class__):
+            return False
+
+        return self.__user_id == other.__user_id  # pylint: disable=protected-access
+
+    def __str__(self) -> str:
+        return str(self.__user_id)
+
+    def __repr__(self) -> str:
+        return repr(self.__user_id)
+
+
+class ResourceId:
+    '''A wrapper for a resource id'''
+
+    def __init__(self, resource_type: Literal['Dataset', 'Layer', 'LayerCollection'],
+                 resource_id: str) -> None:
+        '''Create a resource id'''
+        self.__type = resource_type
+        self.__id = resource_id
+
+    @classmethod
+    def from_layer_id(cls, layer_id: LayerId) -> ResourceId:
+        '''Create a resource id from a layer id'''
+        return ResourceId('Layer', str(layer_id))
+
+    @classmethod
+    def from_layer_collection_id(cls, layer_collection_id: LayerCollectionId) -> ResourceId:
+        '''Create a resource id from a layer collection id'''
+        return ResourceId('LayerCollection', str(layer_collection_id))
+
+    @classmethod
+    def from_dataset_id(cls, dataset_id: DatasetId) -> ResourceId:
+        '''Create a resource id from a dataset id'''
+        return ResourceId('Dataset', str(dataset_id))
+
+    def to_api_dict(self) -> api.ResourceId:
+        '''Convert to a dict for the API'''
+        return {
```

Comment:
return api.ResourceId()

Reply:
why does neither pylint nor mypy catch that?

8d71fe167e49c4484dd2deda9733b5e02562d47f

Reply:
i dont know :(